components/table/table-wrapper.js

```diff
@@ -233,12 +233,14 @@ export class TableWrapper extends RtlMixin(LitElement) {
 		this.noColumnBorder = false;
 		this.stickyHeaders = false;
 		this.type = 'default';
-		this._tableObserver = null;
+		this._tableIntersectionObserver = null;
+		this._tableMutationObserver = null;
 	}
 
 	disconnectedCallback() {
 		super.disconnectedCallback();
-		if (this._tableObserver) this._tableObserver.disconnect();
+		if (this._tableIntersectionObserver) this._tableIntersectionObserver.disconnect();
```

Comment:
There maybe an edge case here that would not work - if this component is disconnected then the observers would be disconnected here, but I don't _think_ they would be reconnected if this component is reconnected to the DOM.  I suspect the `slotchange` event handler would not be rerun on reconnect to the DOM.  In order for this to be a problem, it would have to get disconnected, reconnected, and the table would also have to be hidden.  I am wondering if we even need to disconnect the observers here since the observer and the table being observed are all in this elements subtree.  I certainly feel better if we do though.

Reply:
Yeah agreed... we're definitely inconsistent and maybe a bit over-zealous with this stuff generally. I did some testing with this and you're correct... if I move the `d2l-table-wrapper` DOM node from one place in the document to somewhere else, the observers get disconnected and not re-connected. So I've removed this code that disconnects them altogether. That way when things get re-connected they run again and everything seems to work nicely.